<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChirpConfigWidget</class>
 <widget class="QWidget" name="ChirpConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>744</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,3">
   <item>
    <widget class="ChirpConfigPlot" name="chirpPlot" native="true"/>
   </item>
   <item>
    <widget class="QGroupBox" name="chirpConfigurationBox">
     <property name="title">
      <string>Chirp Configuration</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="addButton">
          <property name="text">
           <string>Add Segment</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="insertButton">
          <property name="text">
           <string>Insert Segment</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="moveUpButton">
          <property name="text">
           <string>Move Up</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="moveDownButton">
          <property name="text">
           <string>Move Down</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="removeButton">
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearButton">
          <property name="text">
           <string>Clear All</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="loadButton">
          <property name="text">
           <string>Load...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableView" name="chirpTable"/>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="preChirpDelayLabel">
          <property name="text">
           <string>Pre-Chirp Delay</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="preChirpDelaySpinBox">
          <property name="toolTip">
           <string>After t0 pulse (which sends TWT enable and protection signals), how long to wait before starting chirp.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string> ns</string>
          </property>
          <property name="minimum">
           <number>300</number>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="protectionDelayLabel">
          <property name="text">
           <string>Protection Delay</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="protectionDelaySpinBox">
          <property name="toolTip">
           <string>Following chirp, how long to wait before closing protection switch.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string> ns</string>
          </property>
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>300</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="chirpsLabel">
          <property name="text">
           <string>Chirps</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="chirpsSpinBox">
          <property name="toolTip">
           <string>Number of chirps in the AWG waveform.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="chirpIntervalLabel">
          <property name="text">
           <string>Chirp Interval</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="chirpIntervalDoubleSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Time from beginning of one chirp to the beginning of the next.&lt;/p&gt;&lt;p&gt;Minimum is pre-chirp delay + chirp duration + protection delay + FID time + 4 Î¼s.&lt;/p&gt;&lt;p&gt;Setting the chirp interval below this value will cause problems, as the scope will miss chirps and become unsynschronized.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string> us</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>5.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>20.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ChirpConfigPlot</class>
   <extends>QWidget</extends>
   <header>chirpconfigplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
